### PR TITLE
feat(controller-sa): Add ability to set extra annotations on the serv…

### PR DESCRIPTION
**Pull request by szalai1**
…iceaccount

This is useful to attach AWS roles to service accounts.

### CODE REVIEW

File: charts/terraform-operator/templates/controller-serviceaccount.yaml [hunk 1]
```diff
@@ -1,10 +1,16 @@
+{{- if .Values.controller.serviceAccount.create }}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: {{ .Release.Name }}
+  name: {{ default .Release.Name }}
```
Comment (isaaguilar) [inline charts/terraform-operator/templates/controller-serviceaccount.yaml]: Why did you include `default` here?

**Review by isaaguilar — COMMENTED** [review]
Changes look good. Just one question about the usage of default.

File: charts/terraform-operator/templates/controller-serviceaccount.yaml (hunk 1 above)
Comment (szalai1) [inline charts/terraform-operator/templates/controller-serviceaccount.yaml]:
```suggestion
  name: {{ .Release.Name }}
```

Comment (szalai1) [reply charts/terraform-operator/templates/controller-serviceaccount.yaml]: removed; wanted to add the option to overwrite, but didn't do it eventually

**Review by isaaguilar — APPROVED** [review]

**Pull request merged**